.github/workflows/build-and-test.yml

```
@@ -87,15 +89,19 @@ jobs:
           text: 'Build failed :sadpanda:'
           username: 'Java-PFB GitHub Action'
 
-  dispatch-tag:
+  tag:
     needs: [ build, unit-tests-and-sonarqube, source-clear ]
-    runs-on: ubuntu-latest
+    uses: ./.github/workflows/tag.yml
+    if: success() && github.ref == 'refs/heads/main'
 
+  publish-library:
+    needs: [ build, unit-tests-and-sonarqube, source-clear, tag ]
```

I'm pretty sure if `tag` needs `build, unit-tests-and-sonarqube, source-clear` and this needs `tag` then it doesn't need to repeat the things `tag` needs.
```suggestion
    needs: [ tag ]
```